### ISSUE TITLE
[TODO] Make LM for ASR

### ISSUE THREAD

**Issue by wannaphong** [PyThaiNLP/pythaiasr#1]
Add lm for ASR

**Comment by tann9949** [comment]
Just some useful information here.

@tanamettpk from ![techcast](https://github.com/TanametTpk) suggested that [pyctcdecode](https://github.com/kensho-technologies/pyctcdecode) is applicable to do a beam search given a sequence of model logits and n-gram LM (.arpa format). 

In my opinion, `pyctcdecode` should work just fine, but I saw many official ASR libraries like NVIDIA's NeMo, deepspeech2, and others use [baidu's ctc-warp](https://github.com/baidu-research/warp-ctc). Therefore, I'm not sure which library should be used for the LM rescoring as `pyctcdecode` might be easier to implement but will definitely be slower that `ctc-warp` as it is implemented in C++.

All in all, I just want to provide you a quick guidelines for your libraries to look for:
- [ ] [pyctcdecode](https://github.com/kensho-technologies/pyctcdecode/tree/main/pyctcdecode)
- [ ] [ctc-warp](https://github.com/baidu-research/warp-ctc)

PS. In case that you work on one of these libraries, please reply to my comment. I would like to implement another library that you will not choose, and let's compare if there's any significant difference.
